swiftwinrt/Resources/Support/ComPtr.swift

```diff
@@ -0,0 +1,100 @@
+// Copyright © 2023 The Browser Company
+// SPDX-License-Identifier: BSD-3
+import C_BINDINGS_MODULE
+
+// ComPtr is a smart pointer for COM interfaces.  It holds onto the underlying pointer
+// and the semantics of it are meant to mirror that of the ComPtr class in WRL. The
+// design of ComPtr and ComPtrs.intialize is that there should be no use of UnsafeMutablePointer
+// any where else in the code base.  The only place where UnsafeMutablePointer should be used is
+// where it's required at the ABI boundary.
+public class ComPtr<CInterface> {

```

Comment:
One unfortunate side-effect of this type is that we lose the in-language representation of nulls

Reply:
it's an optimization to avoid extra addref/release

Reply:
i could change the `CustomQueryInterface` signature to avoid it, but will do that in a different PR